runtime/near-vm-logic/src/logic.rs

```diff
@@ -1481,31 +1484,18 @@ impl<'a> VMLogic<'a> {
         let amount = self.memory_get_u128(amount_ptr)?;
 
         let (receipt_idx, sir) = self.promise_idx_to_receipt_idx_with_sir(promise_idx)?;
-
-        if self.current_protocol_version >= IMPLICIT_ACCOUNT_CREATION_PROTOCOL_VERSION {
-            // Need to check if the receiver_id can be an implicit account.
-            let account_id = self
-                .receipt_to_account
-                .get(&receipt_idx)
-                .expect("promises and receipt_to_account should be consistent.");
-            if is_account_id_64_len_hex(&account_id) {
-                self.gas_counter.pay_action_base(
-                    &self.fees_config.action_creation_config.create_account_cost,
-                    sir,
-                    ActionCosts::transfer,
-                )?;
-                self.gas_counter.pay_action_base(
-                    &self.fees_config.action_creation_config.add_key_cost.full_access_cost,
-                    sir,
-                    ActionCosts::transfer,
-                )?;
-            }
-        }
-        self.gas_counter.pay_action_base(
-            &self.fees_config.action_creation_config.transfer_cost,
-            sir,
-            ActionCosts::transfer,
-        )?;
+        let receiver_id = self.get_account_by_receipt(&receipt_idx);
+        let is_receiver_implicit = self.current_protocol_version
+            >= IMPLICIT_ACCOUNT_CREATION_PROTOCOL_VERSION
```

Comment:
Consider introducing fn is_implicit_account_creation_enabled in version.rs

Reply:
Done